vault/core.go

```diff
@@ -1507,6 +1507,11 @@ func (c *Core) unsealInternal(ctx context.Context, masterKey []byte) (bool, erro
 				c.logger.Warn("failed to notify unsealed status", "error", err)
 			}
 		}
+		if err := c.serviceRegistration.NotifyInitializedStateChange(true); err != nil {
```

Comment:
Question around:

>Ensure that whenever we flag Vault as unsealed, we also flag it as initialized.

Does that mean they indicate the same thing and we could strip one?

There are two potential ways we can think of the definition for "initialized". One is, it's initialized if the Vault operator init command has been executed. The other is, it's initialized if it's up serving traffic. 

What are your thoughts on what our users would expect the meaning of "initialized" to be as it's used by service registration?

Reply:
A Vault server can be:
1. uninitialized and sealed
2. initialized and sealed
3. initialized and unsealed

It cannot be:
4. uninitialized and unsealed

We were already setting the initialized tag (for k8s) when we transition from (1)->(3).  Now we're also setting it when we transition from (2)->(3).

> What are your thoughts on what our users would expect the meaning of "initialized" to be as it's used by service registration?

It should match what `vault status` or the health endpoint reports.

Reply:
Got it! That's helpful, since the term is used in two different ways within Vault. Would it be possible, then, go add that same elaboration to the interface [here](https://github.com/hashicorp/vault/blob/master/serviceregistration/service_registration.go#L87) so future users of the interface will be aware as well?

Reply:
> Got it! That's helpful, since the term is used in two different ways within Vault. 

I only know of one way.  A newly created Vault cluster is uninitialized.  Once `vault operator init` has been run, it is initialized forevermore.  The other meaning you give ("up serving traffic") is the definition of unsealed.  If I had to simplify my rubric from above, the states a Vault core can be in are:

1. uninitialized (cluster-wide)
2. sealed (core-specific)
3. unsealed (core-specific)

> Would it be possible, then, to add that same elaboration to the interface [here](https://github.com/hashicorp/vault/blob/master/serviceregistration/service_registration.go#L87) so future users of the interface will be aware as well?

I added some text, not sure if that covers what you had in mind.

Reply:
I see! I checked out the text and it helps. Thanks!